### PR TITLE
DFA Uniform random word

### PR DESCRIPTION
**Pull request by Tagl**
Added function for uniform random generation for words of length k in DFA

### CODE REVIEW

File: automata/fa/dfa.py [hunk 1]
```diff
@@ -602,6 +603,29 @@ def isfinite(self):
...
+    def random_word(self, k, *, seed=None):
+        self._ensure_count_for_length(k)
+        state = self.initial_state
+        if self._count_cache[k][state] == 0:
+            raise ValueError(f"Language has no words of length {k}")
+
+        result = []
+        rng = Random(seed)
+        for i in range(k):
+            cur_length = k - i
```
Comment (eliotwrobson) [inline automata/fa/dfa.py:615]: I think you can fold this into the range loop, something like `for cur_length in range(k, 1, -1)`.

Comment (Tagl) [reply automata/fa/dfa.py]: Fixed

**Comment by Tagl** [comment]
@caleb531 This one is completely ready, unless you want to add a test that verifies uniformity.

**Review by caleb531 — APPROVED** [review]
@Tagl Looks good to me. Thank you!

**Pull request merged**